src/tools/compiletest/src/runtest/run_make.rs

```diff
@@ -320,35 +296,24 @@ impl TestCx<'_> {
         // To actually run the recipe, we have to provide the recipe with a bunch of information
         // provided through env vars.
 
-        // Compute stage-specific standard library paths.
-        let stage_std_path = host_build_root.join(format!("stage{stage_number}")).join("lib");
-
         // Compute dynamic library search paths for recipes.
+        // These dylib directories are needed to **execute the recipe**.
         let recipe_dylib_search_paths = {
             let mut paths = base_dylib_search_paths.clone();
-
-            // For stage 0, we need to explicitly include the stage0-sysroot libstd dylib.
-            // See <https://github.com/rust-lang/rust/issues/135373>.
-            if std::env::var_os("COMPILETEST_FORCE_STAGE0").is_some() {
-                paths.push(
-                    stage0_sysroot.join("lib").join("rustlib").join(&self.config.host).join("lib"),
-                );
-            }
-
-            paths.push(support_lib_path.parent().unwrap().to_path_buf());
-            paths.push(stage_std_path.join("rustlib").join(&self.config.host).join("lib"));
-            paths
-        };
-
-        // Compute runtime library search paths for recipes. This is target-specific.
-        let target_runtime_dylib_search_paths = {
-            let mut paths = vec![rmake_out_dir.clone()];
-            paths.extend(base_dylib_search_paths.iter().cloned());
+            paths.push(
+                stage0_rustc
+                    .parent()
+                    .unwrap()
+                    .parent()
+                    .unwrap()
+                    .join("lib")
+                    .join("rustlib")
+                    .join(&self.config.host)
+                    .join("lib"),
```

Comment:
Question: maybe pass the stage0 sysroot root instead of doing double parent traversal based on the rustc location? We already assume knowledge of the layout *anyway*. Then maybe above, construct the path to rustc and the host libs here based on that root dir?

Reply:
The [rustc](https://github.com/rust-lang/rust/blob/1c7b60f8a4595efd1cef016582e6050b3cab9868/src/bootstrap/src/core/builder/mod.rs#L1380) function in bootstrap special cases the snapshot compiler, and it gives us a path to the `rustc` executable, not to the sysroot.

So essentially I would just have to do the `parent().unwrap().parent().unwrap()` dance in bootstrap, rather than here. Do you think that's better?

Reply:
Nah, I think this is fine.